scripts/giraffe-facts.py

```diff
@@ -1403,12 +1425,13 @@ def main(args):
     read_lines = read_read_lines(options.vg, options.input, threads=max(1, options.threads // 2))
    
     # Map it to stats and reduce it to total stats
-    stats_total = map_reduce(read_lines, read_line_to_stats, add_in_stats, collections.OrderedDict, threads=max(1, options.threads // 2))
+    stats_total = map_reduce(read_lines, read_line_to_stats, add_in_stats, collections.OrderedDict, 
+                             options.track_last, threads=max(1, options.threads // 2))
```

Comment:
Instead of adding a `map_function_arg` to `map_reduce`, we should use [`functools.partial`](https://docs.python.org/3/library/functools.html#functools.partial) to close over the configuration flag we want to pass to `read_line_to_stats` and make a single-argument function.

```suggestion
    stats_total = map_reduce(read_lines, functools.partial(read_line_to_stats, track_last=options.track_last), add_in_stats, collections.OrderedDict, threads=max(1, options.threads // 2))
```

To get out of having an extra parameter on `read_line_to_stats` to pass through, we could get rid of it and compose `read_line_to_read_dict` and `functools.partial(make_stats, track_last=options.track_last)` right here. And maybe store that in a variable here. But Python doesn't have a handy function composition widget.